server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/mock/util/MockSpamd.java

```diff
@@ -82,16 +96,14 @@ public void run() {
                 out.write(NOT_SPAM.getBytes());
                 out.flush();
             }
-
-            in.close();
-            out.close();
-            spamd.close();
-            socket.close();
-
         } catch (IOException e) {
             // Should not happen
             e.printStackTrace();
+        } finally {
+            IOUtils.closeQuietly(in);
```

Comment:
autocloseable ?

Reply:
I spend 1h debugging after doing that change. Is it essential?

Reply:
I don't understand what is so hard about autocloseable

Reply:
Me neither. As soon as I migrate this code, it breaks. Probably the order in the 'close' clauses.